### ISSUE TITLE
Wibeee Box Mono with firmware 4.4.171, no Sensors showing in Home Assistant.

### ISSUE THREAD

**Issue by povinhox** [luuuis/hass_wibeee#68]
Hi,
I've installed the "hass_wibee" via HACS. I was able to install the Integration and add the device IP and "Disabled (polling only)", also tried the other options. It shows that one entry "Wibeee 7D09A2" and bellow shows "No devices or entries". (see attachment) But unfortunatly I cannot see any sensors. Do I need to create them manually or should they have been populated during the configuration?
It's a Wibeee Box Mono with firmware 4.4.171. I've 3 connectors.
<img width="512" alt="wibeee-1" src="https://github.com/luuuis/hass_wibeee/assets/77890693/2ed00155-c57f-4535-a965-76a158ed586f">
I'm using Home Assistant Container, so it's not easy to edit files in the config folder. 
Any advise?
Thank you.

**Comment by luuuis** [comment]
Hi,

The sensors are discovered automatically from the device.

To allow troubleshooting please [enable debug logging](https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) for the integration. Then reload the integration, download the logs and share them with me.

**Comment by povinhox** [comment]
Hi,
The log file attached.
Thank you for helping.
Best Regards

**Comment by povinhox** [comment]
Hi,
The log file attached.
Thank you for helping.
Best Regards
[home-assistant_wibeee_2023-12-11T22-03-12.203Z.log](https://github.com/luuuis/hass_wibeee/files/13641508/home-assistant_wibeee_2023-12-11T22-03-12.203Z.log)

**Comment by luuuis** [comment]
Ok so the device is not responding to `status.xml` that the integration requires.

```
2023-12-11 22:02:51.778 DEBUG (MainThread) [custom_components.wibeee.api] Error getting http://192.168.1.66/en/status.xml, will retry. ServerDisconnectedError: Server disconnected
```

Please try to restart your Wibeee and then reload the integration @povinhox. Are you able to access the device UI on http://192.168.1.66/ ?

**Comment by povinhox** [comment]
Hi @luuuis,
I've restarted the device and tried again. And again the same result:
`2023-12-12 17:55:19.810 DEBUG (MainThread) [custom_components.wibeee.api] Error getting http://192.168.1.66/en/status.xml, will retry.`
I can access the device login screen and actually log in on it.
I can also access other XML files listed n«in the site I sent you in the previous message. The issue seem to be the "en/status.xml" file.
Am I'm the only one with firmware 4.4.171 having this issue?
<img width="527" alt="WibeeDeviceLoginScreen" src="https://github.com/luuuis/hass_wibeee/assets/77890693/c45dc435-e4ff-42a3-bb22-4aa249e1ad86">
After login:
<img width="941" alt="WibeeUI" src="https://github.com/luuuis/hass_wibeee/assets/77890693/6f9cee3b-fcb6-465c-a91d-c37b94a393b9">
Thank you again,

**Comment by luuuis** [comment]
> Am I the only one with firmware 4.4.171 having this issue?

Maybe you're the only one on this version and using the integration. I, for one, avoid making a firmware upgrade as I'm afraid it will make my Wibeee even more unstable. 

Your screenshot (after login) looks very different to what is provided by my Wibeee on firmware 4.4.169. The old UI was getting data from `status.xml` but perhaps this has changed. Are you able to go into the browser Dev Tools and work out where the data from the screenshot is sourced from?

**Comment by povinhox** [comment]
Hi Luuuis, I got an answer from Wibeee support. It seems that on Firmware version 4.4.171, the data is now on "http://---IP---/services/values2.xml?id=WIBEEE"
I'm adding a xml extract here, but since it shows some private data I'm added the string "-MASKED-" where I've replaced the original values by the string.
I hope it helps.
```
 <values>
<softVersion>4.4.171</softVersion>
<wifiFW>ESP_3.0.3</wifiFW>
<model>WBB</model>
<Bootloader>FALSE</Bootloader>
<spiFlashId>8</spiFlashId>
<BOR>0</BOR>
<POR>1</POR>
<IDLE>0</IDLE>
<SLEEP>0</SLEEP>
<WDTO>0</WDTO>
<SWR>0</SWR>
<EXTR>0</EXTR>
<CM>0</CM>
<IOPWR>0</IOPWR>
<TRAPR>0</TRAPR>
<macAddr>-MASKED-</macAddr>
<serverIp>nest-ingest.wibeee.com</serverIp>
<serverPort>80</serverPort>
<networkType>1</networkType>
<networkTypeText>bss</networkTypeText>
<ipType>1</ipType>
<ipAddr>192.168.1.66</ipAddr>
<gwAddr>192.168.1.254</gwAddr>
<subnetMask>255.255.255.0</subnetMask>
<primaryDNS>192.168.1.254</primaryDNS>
<secondaryDNS>208.67.222.222</secondaryDNS>
<ssid>-MASKED-</ssid>
<bssid/>
<script/>
<keyEnc>-MASKED-</keyEnc>
<keyType>-MASKED-</keyType>
<securKey>-MASKED-</securKey>
<modBus>false</modBus>
<measuresRefresh>60</measuresRefresh>
<appRefresh>1</appRefresh>
<leapThreshold>7</leapThreshold>
<HDataSaveRefresh>1</HDataSaveRefresh>
<clampsModel>-</clampsModel>
<scale>60</scale>
<connectionType>1</connectionType>
<harmonics>1</harmonics>
<enablesqrt>1</enablesqrt>
<invI1>0</invI1>
<minI1>0.15</minI1>
<invI2>0</invI2>
<minI2>0.15</minI2>
<invI3>0</invI3>
<minI3>0.15</minI3>
<irms1>1.08</irms1>
<ithd1>82.50</ithd1>
<iharm11>1.00</iharm11>
<iharm31>0.70</iharm31>
<iharm51>0.40</iharm51>
<iharm71>0.20</iharm71>
<iharm91>0.00</iharm91>
<vrms1>236.05</vrms1>
<vthd1>0.00</vthd1>
<vharm11>235.30</vharm11>
<vharm31>0.00</vharm31>
<vharm51>0.00</vharm51>
<vharm71>0.00</vharm71>
<vharm91>0.00</vharm91>
<angle1>0.00</angle1>
<freq1>50.15</freq1>
<pap1>255.71</pap1>
<pac1>202.24</pac1>
<preac1>-150.48</preac1>
<fpot1>0.791</fpot1>
<eac1>41</eac1>
<eacprod1>0</eacprod1>
<eaccons1>41</eaccons1>
<eacq11>0</eacq11>
<eacq21>0</eacq21>
<eacq31>0</eacq31>
<eacq41>41</eacq41>
<ereac1>0</ereac1>
<ereacl1>0</ereacl1>
<ereacc1>9</ereacc1>
<ereacq11>0</ereacq11>
<ereacq21>0</ereacq21>
<ereacq31>0</ereacq31>
<ereacq41>9</ereacq41>
<irms2>0.00</irms2>
<ithd2>0.00</ithd2>
<iharm12>0.00</iharm12>
<iharm32>0.00</iharm32>
<iharm52>0.00</iharm52>
<iharm72>0.00</iharm72>
<iharm92>0.00</iharm92>
<vrms2>236.05</vrms2>
<vthd2>0.00</vthd2>
<vharm12>235.30</vharm12>
<vharm32>0.00</vharm32>
<vharm52>0.00</vharm52>
<vharm72>0.00</vharm72>
<vharm92>0.00</vharm92>
<freq2>50.15</freq2>
<angle2>0.00</angle2>
<pap2>0.00</pap2>
<pac2>0.00</pac2>
<preac2>0.00</preac2>
<fpot2>0.000</fpot2>
<eac2>0</eac2>
<eacprod2>0</eacprod2>
<eaccons2>0</eaccons2>
<eacq12>0</eacq12>
<eacq22>0</eacq22>
<eacq32>0</eacq32>
<eacq42>0</eacq42>
<ereac2>0</ereac2>
<ereacl2>0</ereacl2>
<ereacc2>0</ereacc2>
<ereacq12>0</ereacq12>
<ereacq22>0</ereacq22>
<ereacq32>0</ereacq32>
<ereacq42>0</ereacq42>
<irms3>0.00</irms3>
<ithd3>0.00</ithd3>
<iharm13>0.00</iharm13>
<iharm33>0.00</iharm33>
<iharm53>0.00</iharm53>
<iharm73>0.00</iharm73>
<iharm93>0.00</iharm93>
<vrms3>236.05</vrms3>
<vthd3>0.00</vthd3>
<vharm13>235.30</vharm13>
<vharm33>0.00</vharm33>
<vharm53>0.00</vharm53>
<vharm73>0.00</vharm73>
<vharm93>0.00</vharm93>
<angle3>0.00</angle3>
<freq3>50.15</freq3>
<pap3>0.00</pap3>
<pac3>0.00</pac3>
<preac3>0.00</preac3>
<fpot3>0.000</fpot3>
<eac3>0</eac3>
<eacprod3>5</eacprod3>
<eaccons3>0</eaccons3>
<eacq13>0</eacq13>
<eacq23>0</eacq23>
<eacq33>5</eacq33>
<eacq43>0</eacq43>
<ereac3>0</ereac3>
<ereacl3>0</ereacl3>
<ereacc3>0</ereacc3>
<ereacq13>0</ereacq13>
<ereacq23>0</ereacq23>
<ereacq33>0</ereacq33>
<ereacq43>0</ereacq43>
<irmst>1.15</irmst>
<vrmst>235.93</vrmst>
<freqt>50.00</freqt>
<papt>271.58</papt>
<pact>177.68</pact>
<preact>-145.46</preact>
<fpott>0.654</fpott>
<eact>35</eact>
<eacprodt>6</eacprodt>
<eacconst>41</eacconst>
<eacq1t>0</eacq1t>
<eacq2t>0</eacq2t>
<eacq3t>6</eacq3t>
<eacq4t>41</eacq4t>
<ereact>0</ereact>
<ereaclt>0</ereaclt>
<ereacct>9</ereacct>
<ereacq1t>0</ereacq1t>
<ereacq2t>0</ereacq2t>
<ereacq3t>0</ereacq3t>
<ereacq4t>9</ereacq4t>
</values>
```

**Comment by luuuis** [comment]
Thanks @povinhox. Looks like Wibeee has come up with a third format for reading the data. After`status.xml` and `values.xml`, we now have `values2.xml` with slightly different format.

I will look into adding support for the new firmware. I think I have enough info at the moment but will need you to test on your device once that is available.

**Comment by povinhox** [comment]
Happy to help! 
Let me know if I need to try any beta version of the integration. 
After asking Wibeee / Similics support about the meaning of each variable in the new XML, and explaining that their page https://support.wibeee.com/space/CH/184025089/XML is not up to date, they told me that they intend to update the page, but for now they asked me what specific variables I want to know. So If you need my help I can ask them, if not you know they are willing to support on that.
Once Again Thank you / Obrigado!

**Comment by luuuis** [comment]
Please try https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.0 and report back.

**Comment by ega6226** [comment]
Hello,

Do I have to replace the current folder with the new one? Is there anything else to do?

Thanks

**Comment by luuuis** [comment]
If using HACS, the preferred install method, you need to [enable beta versions](https://hacs.xyz/docs/faq/beta/) to upgrade to `v3.4.0-beta.0`.

If not using HACS then replace the contents of `custom_components/wibeee` with the new release.

**Comment by ega6226** [comment]
I have replaced the folder and restarted HA, still the same.

**Comment by ega6226** [comment]
![image](https://github.com/luuuis/hass_wibeee/assets/43537728/9c0953c4-6e4f-4f3d-8c66-3aec98177c34)

**Comment by luuuis** [comment]
@ega6226 if you wish to help and are able, check https://github.com/luuuis/hass_wibeee/issues/68#issuecomment-1849680694 for how to provide useful troubleshooting info.

⚠️ Wibeee has decided that it was a good idea to include something like `<securKey>YOUR_WIFI_PASSWORD</securKey>` in the device response. Before sharing anything here you should definitely review the log file and remove your WiFi password at least. Alternatively, email me the file at cheery.key9586@fastmail.com.

**Comment by luuuis** [comment]
Thanks @ega6226, I received your log. I see the following problem:

```
2023-12-19 12:34:38.953 DEBUG (MainThread) [custom_components.wibeee.api] Error getting http://192.168.1.40/services/values2.xml?id=WIBEEE, will retry. ServerDisconnectedError: Server disconnected
```

It seems that `/services/values2.xml?id=WIBEEE` is maybe the wrong address for the new API. I don't have this firmware so I was going from the information in https://github.com/luuuis/hass_wibeee/issues/68#issuecomment-1854440605.

Can you please try in your browser to open http://192.168.1.40/services/user/values2.xml?id=WIBEEE and see if you get a response with the meter's values?

**Comment by luuuis** [comment]
Thanks again @ega6226. Please check https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.1 which now uses the correct API address.

**Comment by povinhox** [comment]
Hi,
Ive tried the v3.04-beta.1 and didn't got any sensors, like last time. I'm attaching the log with some masked values. 
[home-assistant_wibeee_2023-12-19T14-02-13.036Z.log](https://github.com/luuuis/hass_wibeee/files/13716568/home-assistant_wibeee_2023-12-19T14-02-13.036Z.log)

**Comment by luuuis** [comment]
Another small difference in `values2.xml` compared to `status.xml`. Please try https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.2

**Comment by povinhox** [comment]
Still no luck 
[home-assistant_wibeee_2023-12-19T14-34-48.510Z.log](https://github.com/luuuis/hass_wibeee/files/13716845/home-assistant_wibeee_2023-12-19T14-34-48.510Z.log)

**Comment by ega6226** [comment]
It does not work for me.

**Comment by luuuis** [comment]
Right, I think this should finally do it. Thank you for your patience. 

https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.3

**Comment by povinhox** [comment]
Success! Thank you very much! Muito Obrigado!
<img width="891" alt="Captura de ecrã 2023-12-19 154234" src="https://github.com/luuuis/hass_wibeee/assets/77890693/a696589d-3ed6-4964-b2b5-4424cb689e38">
<img width="548" alt="Captura de ecrã 2023-12-19 154336-2" src="https://github.com/luuuis/hass_wibeee/assets/77890693/11b00eac-9b6b-4247-a2a8-0a3520058a2f">

**Comment by luuuis** [comment]
Excellent! I have one last change and that is to avoid getting back WiFi passwords and extra data in the response (this will make support easier in the future and prevent mistakes).

Could you install https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.4 and verify that all works as before (it should if `values2.xml` works the same as the documented `values.xml`). If all is well I will then release v3.4.0.

**Comment by ega6226** [comment]
I will test it tomorrow and report back.

**Comment by povinhox** [comment]
With beta 4, it's not working. 
[home-assistant_wibeee_2023-12-19T22-45-15.838Z.log](https://github.com/luuuis/hass_wibeee/files/13721074/home-assistant_wibeee_2023-12-19T22-45-15.838Z.log)

**Comment by luuuis** [comment]
Ok, a silly oversight was the cause for that and was fixed in [v3.4.0-beta.5](https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0-beta.5). If this works please add logs so I can verify that WiFi secrets are no longer loggewd @povinhox.

Also, can you confirm that there is continuity between the sensors of the 3.3.x integration and the current ones? For example, if you look at the history for "active power L1" it should be the same sensor that existed before the upgrade and should keep all the history (recorder and statistics). This should be the case except for the reactive power sensor, which is exposed differently in the `values2` API.

**Comment by povinhox** [comment]
Hi Luis, 
Unfortunately didn't work for me. I've also tried to remove and add the Integration, an with that all sensor are gone. I've also restarted the Wibeee device since I saw a lot of retries. I've tried to open the values2.xml via browser and I can see the file. 
Log attached.
[home-assistant_wibeee_2023-12-20T17-37-33.709Z-beta5.log](https://github.com/luuuis/hass_wibeee/files/13730460/home-assistant_wibeee_2023-12-20T17-37-33.709Z-beta5.log)

**Comment by luuuis** [comment]
The fix was released in [v3.4.0](https://github.com/luuuis/hass_wibeee/releases/tag/v3.4.0). I will handle the WiFi secrets logging separately.

**Comment by povinhox** [comment]
With 3.4.0 It's working.  Thank you for all your hard work.
It seems to get the wifi password into the log. As before the beta 4.
As for if there is continuity between the sensors of the 3.3.x integration and the current ones. I cannot since I already had the firmware 4.4.171 when i first installed the integration. So I have no data from before.
